code/controllers/subsystems/holomap.dm

```diff
@@ -0,0 +1,113 @@
+// Minimap generation system adapted from vorestation, adapted from /vg/.
+// Seems to be much simpler/saner than /vg/'s implementation.
+
+//The structure holding a given map
+/datum/holomapdata
+	var/icon/holomap_base
+	var/list/icon/holomap_areas = list()
+	var/icon/holomap_combined
+	var/icon/holomap_areas_combined
+	var/icon/holomap_small
+
+SUBSYSTEM_DEF(minimap)
+	name = "Holomap"
+	flags = SS_NO_FIRE
+	init_order = SS_INIT_HOLOMAP
+
+	var/list/datum/holomapdata/holomaps = list()
+	var/list/station_holomaps = list()
+
+/datum/controller/subsystem/minimap/Initialize()
+	holomaps.len = world.maxz
+	for (var/z = 1 to world.maxz)
+		generateHolomap(z)
+
+	//Update machinery if it has not been
+	for(var/obj/machinery/holomap/M in station_holomaps)
+		if(istype(M) && !QDELETED(M))
```

Comment:
Doesn't make sense; `istype` always passes and if it's qdeleted and still in there you've got bigger problems.